x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -206,29 +191,8 @@ private static Object innerDeepCopy(Object value, boolean unmodifiable) {
         }
     }
 
-    private static class CacheKey {
-
-        final String enrichIndex;
-        final SearchRequest searchRequest;
-
-        private CacheKey(String enrichIndex, SearchRequest searchRequest) {
-            this.enrichIndex = enrichIndex;
-            this.searchRequest = searchRequest;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            CacheKey cacheKey = (CacheKey) o;
-            return enrichIndex.equals(cacheKey.enrichIndex) && searchRequest.equals(cacheKey.searchRequest);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(enrichIndex, searchRequest);
-        }
-    }
+    // Visibility for testing
+    record CacheKey(String enrichIndex, int maxMatches, Object value) {}
```

Comment:
The original cache key (the `SearchRequest`) took these three values as parameters:
https://github.com/elastic/elasticsearch/blob/1ee0be8ba821bc076efa8cae880f0f343f23ae4e/x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java#L71-L82
Technically, it also took the `matchField` as a parameter, but that is fixed/embedded into the enrich index so we don't need to include it explicitly.